app/models/ems_event.rb

```diff
@@ -136,6 +137,18 @@ def self.process_host_in_event!(event, options = {})
     end
   end
 
+  def self.process_storage_in_event!(event, options = {})
+    ems_ref = event[:ems_ref]
+    process_object_in_event!(PhysicalStorage, event, options)
+
+    if event[:physical_storage_id].nil? && ems_ref.present?
+      storage = PhysicalStorage.find_by(:ems_ref => ems_ref)
+      unless storage.nil?
+        event[:physical_storage_id] = storage.id
+      end
+    end
```

Comment:
I don't think this is necessary, if we weren't able to find the physical_storage by ems_id+ems_ref then just looking by ems_ref is asking to find one that is in another provider which is incorrect.
```suggestion
```